src/libertem/io/dataset/base/backend.py

```diff
@@ -75,10 +77,10 @@ def need_copy(self, roi, native_dtype, read_dtype, tiling_scheme=None, fileset=N
             log.debug("have decode, need copy")
             return True
 
-        # 3) if we have less frames per file than tile depth, we need to copy, too
+        # 3) if we have less or same frames per file than tile depth, we need to copy, too
```

Comment:
Why do we need to copy if the tile depth exactly matches the number of frames per file? I'm not sure I follow here.

Reply:
In case of `DM` files, when a shape < number of files was given, the tile depth is sometimes equal to number of frames per file and `get_tiles_straight` was used instead of `get_tiles_w_copy`.

Reply:
> the tile depth is sometimes equal to number of frames per file and `get_tiles_straight` was used instead of `get_tiles_w_copy`.

Hmm, why is it a problem to use `_get_tiles_straight` in that situation? It is only used if we don't need to do any additional decoding (i.e. `dtype` matches, no corrections need to be applied etc.), so it should be fine to be used, no? Saving a copy can be very beneficial to performance.

Are there any other reasons we need to do a copy, which are not currently covered by `need_copy`?

Reply:
> Hmm, why is it a problem to use `_get_tiles_straight` in that situation?

In the case mentioned above, `_get_tiles_straight` yielded empty tiles (0 in the tile_slice shape).

> It is only used if we don't need to do any additional decoding (i.e. `dtype` matches, no corrections need to be applied etc.), so it should be fine to be used, no?

That sounds right, I think there's a bug in `DM`'s `fileset` part of code. I'll check.

> Are there any other reasons we need to do a copy, which are not currently covered by `need_copy`?

Not sure, will look into it.